tests/jerry/array-prototype-splice.js

```diff
@@ -175,3 +175,203 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+try {
+  Array.prototype.splice.call(undefined);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("foo"); } });
+  var a = { length : o };
+  Array.prototype.splice.call(a, function() { });
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "foo");
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("1"); } });
+  [1, 2].splice(o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "1");
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("2"); } });
+  [1, 2].splice(1, o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "2");
+}
+
+try {
+  var a = [1, 5, ,2];
+  Object.defineProperty(a, '2', { 'get' : function() { throw new ReferenceError("3"); } });
+  a.splice(1, 7, 7, 7, 7, 7);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "3");
+}
+
+var a = {length : 13};
+Array.prototype.splice.call(a, function () {delete a});
+assert(a.length === 0);
+
+try {
+  var a = [1, 5, 6, 7, 8, 5];
+  Object.defineProperty(a, '0', { 'get' : function() { throw new ReferenceError("foo0"); } });
+  Array.prototype.splice.call(a, 0, 3);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+  assert(e.message == "foo0");
+}
+
+try {
+  f = function() {delete arr[3]; arr.length = 13; Object.defineProperty(arr, '5', function() { })};
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  Object.defineProperty(arr, '2',{ 'get' : f } );
+  for(var i = 0; i < arr.length; i++)
+  {
+    var a = arr[i];
+  }
+  arr.splice(1, 4, obj);
+} catch (e) {}
+
+try {
+  f = function() { for(var i = 0; i < arr.length; i++) { delete arr[i]} };
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  Object.defineProperty(arr, '0', { 'get' : f });
+  Object.defineProperty(arr, '1', { 'get' : f });
+  Object.defineProperty(arr, '2', { 'get' : f });
+  Object.defineProperty(arr, '3', { 'get' : f });
+  Object.defineProperty(arr, '4', { 'get' : f });
+  Object.defineProperty(arr, '5', { 'get' : f });
+  arr.splice(1, 3, obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+try {
+  f = function() { throw new TypeError("4");};
+  obj = {get: f, valueOf : f, toString: f};
+  arr = [1, 2, obj, 4, 5];
+  Object.defineProperty(arr, '4', { 'get' : f });
+  arr.splice(1, 3, obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+  assert(e.message == "4");
+}
+
+try{
```

Comment:
Could you please add a short description for each new testcase? After this point the test cases are slightly complicated, it would be good to know what is happening.